src/main/java/moze_intel/projecte/emc/NormalizedSimpleStack.java

```diff
@@ -254,15 +273,23 @@ public String toString() {
 			return "OD: " + od;
 		}
 	}
-
+	//TODO Add NBT support here
 	public static NormalizedSimpleStack fromSerializedItem(String serializedItem) {
 		int pipeIndex = serializedItem.lastIndexOf('|');
+		int atIndex = serializedItem.lastIndexOf('@');
 		if (pipeIndex < 0)
 		{
 			throw new IllegalArgumentException(String.format("Cannot parse '%s' as itemstack. Missing | to separate metadata.", serializedItem));
 		}
 		String itemName = serializedItem.substring(0, pipeIndex);
-		String itemDamageString = serializedItem.substring(pipeIndex + 1);
+		String itemDamageString;
+		String itemNBTString = null;
+		if(atIndex < 0){
+			itemDamageString = serializedItem.substring(pipeIndex + 1);
+		} else{
+			itemDamageString = serializedItem.substring(pipeIndex + 1, atIndex);
+			itemNBTString = serializedItem.substring(atIndex + 1);
+		}
```

Comment:
Could use `JsonToNBT` in the minecraft source here. That will do the deserialize for you.